vaadin-simple-grid-filter/src/main/java/software/xdev/vaadin/FilterComponent.java

```diff
@@ -515,8 +516,23 @@ private void deactivateDeleteButtonFromChipComponents(
 					this.addQueryParameter(badge);
 				}
 				
-				// Activate the reset button
-				this.btnResetFilter.setEnabled(true);
+				// When no initial filter is existing
+				if(this.initialChipBadges.isEmpty() && this.chipBadges.isEmpty())
+				{
+					this.btnResetFilter.setEnabled(false);
+				}
+				else
+				{
+					final List<ChipBadgeExtension<FilterCondition<T, ?>>> initialChipBadgesCopy
+						= new ArrayList<>(this.initialChipBadges);
+					final List<ChipBadgeExtension<FilterCondition<T, ?>>> chipBadgesCopy
+						= new ArrayList<>(this.chipBadges);
+					
+					// Check if just the initial filter are currently applied. Then enable/disable the reset button as
+					// appropriate.
+					this.btnResetFilter.setEnabled(
+						!new FilterComponentUtl<T>().equalLists(initialChipBadgesCopy, chipBadgesCopy));
```

Comment:
I think the comparison can be done a in a simpler way using something like this:

```java
this.initialChipBadges.stream()
	.map(ChipBadge::getBadgeId)
	.collect(Collectors.toSet())
	.equals(this.chipBadges.stream()
		.map(ChipBadge::getBadgeId)
		.collect(Collectors.toSet()))
```